public/scripts/api-helpers.js

```diff
@@ -131,4 +117,71 @@ const callAPIByCategory = async (task) => {
     default:
       return task;
   }
-}
+};
+
+/**
+ * Gives a time limit to a promise
+ * Given promise is rejected if not resolved before timeout
+ * @param  {number}      ms      Length of ms before timeout
+ * @param  {Promise<{}>} promise Promise to limit
+ * @return {Promise<>}
+ */
+const timeoutPromise = (ms, promise) => {
+  const timeout = new Promise((resolve, reject) =>
+    setTimeout(() =>
+      reject(`Timed out after ${ms} ms.`), ms));
+
+  return Promise.race([
+    promise,
+    timeout
+  ]);
+};
+
+/**
+ * Checks if given word is found in the given string
+ * @param  {string}  str  String to check condition
+ * @param  {string}  word Word to look for in the string
+ * @return {boolean}      Boolean if word found in string
+ */
+const findWordInString = (str, word) => {
+  return new RegExp('\\b' + word + '\\b', 'i').test(str);
+};
+
+/**
+ * Matches string against key values to check for easy API match
+ * @param  {string} string Takes in string to check for keyword
+ * @return {string}        String of category matched, else undefined
+ */
+const matchCategoryKeyword = (string) => {
+  let category;
+  const categories = ['restaurants', 'films', 'books', 'products'];
+  const keywords = ['eat', 'watch', 'read', 'buy'];
+
+  keywords.some((val, index) => {
+    if (findWordInString(string, val)) {
+      category = categories[index];
+    }
+  });
+
+  return category;
+};
+
+/**
+ * Trim given string of matched keyword and whitespace from the start/end
+ * @param  {string} string Takes in string to check for keyword
+ * @return {string}        Trimmed string
+ */
+const filterKeywords = (string) => {
+  const keywords = ['eat', 'watch', 'read', 'buy'];
+  let newString = '';
+
+  keywords.some((val, index) => {
+    if (findWordInString(string, val)) {
+      newString = string.replace(val, '').trim();
```

Comment:
I realized the real problem here: it loops through the array (since the callback has no return value it executes for each keyword) and filters out the keyword, BUT each iteration newString is overwritten since replace is ran on string, not newString, so it only filters out the last keyword